src/Worker.Extensions.DurableTask/FunctionsOrchestrationContext.InputConverter.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Text.Json;
+using Microsoft.DurableTask;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions.DurableTask;
+
+internal sealed partial class FunctionsOrchestrationContext
+{
+    private abstract class InputConverter
+    {
+        public abstract T Get<T>();
+
+        public static InputConverter Create(object? baseInput, DataConverter converter)
+        {
+            return baseInput switch
+            {
+                JsonElement json => new JsonElementInputConverter(json),
+                null => NullInputConverter.Instance,
+                _ => new DefaultInputConverter(baseInput, converter),
+            };
+        }
+    }
+
+    private class DefaultInputConverter : InputConverter
+    {
+        private readonly DataConverter converter;
+        private readonly string serializedInput;
+
+        public DefaultInputConverter(object input, DataConverter converter)
+        {
+            this.converter = converter;
+            this.serializedInput = converter.Serialize(input);
```

Comment:
Are we going to be doing this for base types, like `string`, `int`, etc.?  We didn't have to previously, right?

Reply:
Previously base types did not work at all either. You would get an error like `error CS0030: Cannot convert type 'System.Text.Json.JsonElement' to 'string'`

However, I have a check in `GetInput<T>` which checks if the base type casts directly to the requested type and I return it directly.